src/index.js

```diff
@@ -98,7 +99,206 @@ class Griddle extends Component {
 
     this.events = Object.assign({}, events, ...plugins.map(p => p.events));
 
-    this.selectors = plugins.reduce((combined, plugin) => ({ ...combined, ...plugin.selectors }), {...selectors});
```

Comment:
Before we dig into reviewing this code, can you pull your whole new block into `./utils/selectorUtils.js` to minimize the churn here? Something like:

```
this.selectors = flattenSelectors(selectors, plugins);
```

That should make it a lot easier to test your selector composition logic in isolation.

(Please squash this change, to avoid merge conflicts in this file.)